### PR TITLE
Refactor check.py into seperate functions.  NFC.

### PR DESCRIPTION
**Pull request by sbc100**
Once could argue that they should go in separate files
but this seems like a good first step.

This is intended to be a non-functional-change.

### CODE REVIEW

**Comment by sbc100** [comment]
As a followup it should be possible to use to the command line to select which test(s) to run

**Comment by sbc100** [comment]
Github has done a terribel job of formatting this change..  It mostly just indentation changes.

File: check.py [hunk 1]
```diff
@@ -698,6 +697,32 @@ def execute():
               shutil.copyfile(recreated, 'a.wasm.wast')
               execute()
 
+
+# Run all the tests
+run_help_tests()
+run_wasm_opt_tests()
+s2wasm.test_s2wasm()
```
Comment (jgravelle-google) [inline check.py]: This looks like it changes the order of what gets ran when. I'm not inherently opposed to that, but I'm not sure if the existing order had some broader reason behind it.

Comment (sbc100) [reply check.py]: Oops. Fixed.

**Comment by binji** [comment]
The PR is more readable like this (no whitespace in diffs): https://github.com/WebAssembly/binaryen/pull/1158/files?w=1

**Pull request merged**